### PR TITLE
PartialFunction polishing

### PR DESCRIPTION
**Pull request by pavelpavlov**
- ScalaDocs added
- TODOs fixed
- controversive method `run` deleted
- not used class runtime.AbstractTotalFunction removed
- small corrections & fixes
- tests for `orElse` & `runWith`

review by @adriaanm

### CODE REVIEW

**Comment by jsuereth** [comment]
So, just want to say the documentation looks great.  I need to think through the code changes a bit before commenting.  I can't focus right now since I'm in a meeting :)

File: src/library/scala/PartialFunction.scala [hunk 1]
```diff
@@ -150,23 +173,40 @@ object PartialFunction {
...
+    def apply(x: A): C = k(pf(x))
 
-  private[scala] final class Lifted[-A, +B] (val pf: PartialFunction[A, B])
+    override def applyOrElse[A1 <: A, C1 >: C](x: A1, default: A1 => C1): C1 = {
+      val z = pf.applyOrElse(x, checkFallback[B])
+      if (!fallbackOccurred(z)) k(z) else default(x)
+    }
+  }
+
+  private[this] final val fallback_pf: PartialFunction[Any, Any] = { case _ => fallback_pf }
```
Comment (retronym) [inline src/library/scala/PartialFunction.scala:210]: Could you add some comments on the role of `fallback_pf` in the implementation?

Comment (pavelpavlov) [inline src/library/scala/PartialFunction.scala:210]: Sure. Should I apend a commit to this pull request or create another one?

Comment (xeno-by) [inline src/library/scala/PartialFunction.scala:210]: I typically maintain a single commit on top of the pull request named "pull request feedback", squashing new feedback into that commit if necessary.

Comment (pavelpavlov) [inline src/library/scala/PartialFunction.scala:210]: fixed

**Comment by retronym** [comment]
You could skip the rebuild for the addition of the comment; just delete your comment before the bot notices it.

**Comment by paulp** [comment]
Looks clever/good to me.

**Comment by jsuereth** [comment]
PLS REBUILD ALL

**Comment by gkossakowski** [comment]
LGTM from me as well. It's very clever and nicely documented code. Good job @pavelpavlov!

**Pull request merged**